src/components/formComponent/FormComponent.jsx

```diff
@@ -0,0 +1,288 @@
+	});
+
+	const { fields, append, remove } = useFieldArray({
+		control,
+		name: 'experience',
+	});
+
+	const onSubmit = (data) => {
+		setIsFormSubmitted(true);
+		setUserCv(URL.createObjectURL(data.cv[0]));
+		setUserData(data);
+		reset();
+	};
+
+	return (
+		<>
+			<h1>Formularz zgłoszeniowy na kurs programowania</h1>
+			<form
+				className={className}
+				onSubmit={handleSubmit(onSubmit)}
+				encType='multipart/form-data'
+			>
+				<FormSectionComponent headerText='Dane osobowe'>
+					<InputComponent
+						name='firstName'
+						placeholder='Imię'
+						register={register}
+						classNameError={style.error}
+						error={errors.firstName}
+					/>
+					<InputComponent
+						name='lastName'
+						placeholder='Nazwisko'
+						register={register}
+						classNameError={style.error}
+						error={errors.lastName}
+					/>
+					<InputComponent
+						name='email'
+						placeholder='E-mail'
+						register={register}
+						classNameError={style.error}
+						error={errors.email}
+					/>
+					<InputComponent
+						type='number'
+						name='phoneNumber'
+						placeholder='Numer telefonu'
+						register={register}
+						error={errors.phoneNumber}
+						classNameError={style.error}
+					/>
+				</FormSectionComponent>
+				<FormSectionComponent headerText='Preferencje Kursu'>
+					<div className={style.row}>
+						<p>Wybierz formę nauki:</p>
+						<div>
+							<InputComponent
+								type='radio'
+								name='preferences'
+								value='stacjonarny'
+								id='stationary'
+								register={register}
+								defaultChecked
+							/>
+							<label htmlFor='stationary'>Stacjonarne</label>
+						</div>
+						<div>
+							<InputComponent
+								type='radio'
+								name='preferences'
+								value='online'
+								id='online'
+								register={register}
+							/>
+							<label htmlFor='online'>Online</label>
+						</div>
+					</div>
+					<SelectComponent
+						name='prefTechnology'
+						className={style.select_pref_tech}
+						size={5}
+						multiple
+						register={register}
+						options={[
+							{ value: 'React', text: 'React' },
+							{ value: 'Node.js', text: 'Node.js' },
+							{ value: 'CSS', text: 'CSS' },
+							{ value: 'HTML', text: 'HTML' },
+							{ value: 'Next.js', text: 'Next.js' },
+						]}
+					/>
+					{errors.prefTechnology && (
+						<span className={style.error}>{errors.prefTechnology.message}</span>
+					)}
+				</FormSectionComponent>
+				<FormSectionComponent headerText='Dodaj swoje CV'>
+					<InputComponent
+						type='file'
+						name='cv'
+						register={register}
+						error={errors.cv}
+						classNameError={style.error}
+					/>
+				</FormSectionComponent>
+				<FormSectionComponent headerText='Doświadczenie w programowaniu'>
+					<div className={style.row}>
+						<InputComponent
+							type='checkbox'
+							name='experienceCheckbox'
+							register={register}
+							onClick={() => setIsExperienceChecked(!isExperienceChecked)}
```

Comment:
When operating on the previous state, we usually use the notation prev => !prev.